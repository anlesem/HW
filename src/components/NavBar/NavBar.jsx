import React from 'react';
import { NavLink } from 'react-router-dom';

import style from './NavBar.module.scss';

const navigate = [
   {
      id: 1,
      to: '/',
      name: 'Профиль',
   },
   {
      id: 2,
      to: '/chats',
      name: 'Чаты',
   }
];

export const NavBar = () => {
   return (
      <ul className={style.navbar}>
         {navigate.map((link) => {
            return (
               <li key={link.id}>
                  <NavLink
                     to={link.to}
                     className={({ isActive }) => isActive ? style.active : ''}
                     data-testid={`NavLink-${link.id}`}>
                     {link.name}
                  </NavLink>
               </li>
            );
         })}
      </ul>
   );
};