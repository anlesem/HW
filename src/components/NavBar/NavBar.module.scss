.navbar {
   display: flex;
   justify-content: space-around;
   width: 100%;
}

.active {
   font-weight: bold;
}