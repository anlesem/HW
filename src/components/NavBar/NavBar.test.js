import { BrowserRouter } from 'react-router-dom';
import { render, screen } from '@testing-library/react';

import { NavBar } from './NavBar';

describe('NavBar', () => {
   it('Компонент существует', () => {
      expect(NavBar).toBeInstanceOf(Function);
   });

   it('Изначально список сообщений существует, но пустой', () => {
      render(<BrowserRouter><NavBar /></BrowserRouter>);

      expect(screen.getByRole('list')).toBeTruthy();
      expect(screen.getAllByRole('listitem')).toHaveLength(2);
   });
});