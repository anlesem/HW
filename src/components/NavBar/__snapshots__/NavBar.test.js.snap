// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar Снимок состояния 1`] = `
<DocumentFragment>
  <ul
    class="navbar"
  >
    <li>
      <a
        aria-current="page"
        class="active"
        data-testid="NavLink-1"
        href="/"
      >
        Профиль
      </a>
    </li>
    <li>
      <a
        class=""
        data-testid="NavLink-2"
        href="/chats"
      >
        Чаты
      </a>
    </li>
  </ul>
</DocumentFragment>
`;
